x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats.csv-spec

```diff
@@ -1697,3 +1697,60 @@ FROM employees | STATS min = min(salary) by languages | SORT min + CASE(language
 29175          |2              
 28336          |null           
 ;
+
+
+weightedAvg
+required_capability: weighted_avg
+from employees
+| stats w_avg_1 = weighted_avg(salary, 1), avg = avg(salary), w_avg_2 = weighted_avg(salary, height)
+| EVAL w_avg_1 = ROUND(w_avg_1), avg = ROUND(avg), w_avg_2 = ROUND(w_avg_2)
+;
+
+w_avg_1:double | avg:double | w_avg_2:double
+48249.0        | 48249.0   | 48474.0
+;
+
+weightedAvgGrouping
+required_capability: weighted_avg
+// tag::weighted-avg[]
+FROM employees
+| STATS w_avg = WEIGHTED_AVG(salary, height) by languages
+| EVAL w_avg = ROUND(w_avg)
+| KEEP w_avg, languages
+| SORT languages
+// end::weighted-avg[]
+;
+
+// tag::weighted-avg-result[]
+w_avg:double | languages:integer
+51464.0      | 1
+48477.0      | 2
+52379.0      | 3
+47990.0      | 4
+42119.0      | 5
+52142.0      | null
+// end::weighted-avg-result[]
+;
+
+weightedAvgConstant
+required_capability: weighted_avg
+row v = [1, 2, 3]
+| stats w_avg_1 = weighted_avg(v, 1), w_avg_2 = weighted_avg([1, 2, 3], 1), avg = avg(v)
+| EVAL w_avg_1 = ROUND(w_avg_1), w_avg_2 = ROUND(w_avg_2), avg = ROUND(avg)
+;
+
+w_avg_1:double |w_avg_2:double |avg:double
+2.0            | 2.0           | 2.0
+;
+
+weightedAvgConstantMvWarning
+required_capability: weighted_avg
+row v = [1, 2, 3], w = [1, 2, 3]
```

Comment:
This is good!

Additionally, I think we could use some tests where
- the weights are constant, multi-valued but the field is not and vice versa.
- the field/weights are null
- the field/weights are an expression that folds to null